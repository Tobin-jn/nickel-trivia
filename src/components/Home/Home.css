html, body {
  width: 100%;
  height: 100%;
}

.landing-page {
  height: 100vh;
  background: linear-gradient(to bottom, rgba(255, 255, 255, 1), rgba(255, 255, 255, 0)), url(../../images/blanket.jpg) center center no-repeat;
  background-size: cover;
}

.title {
  text-align: center;
  color: red;
  position: absolute;
  top: 200px;
}

.nickel-logo {
  position: absolute;
  top: 150px;
  left: 38%;
  width: 300px;
}

.btn-container {
  display: flex;
  flex-direction: row;
  justify-content: center;
  position: relative;
  top: 500px;
}

.btn {
  border: black thin solid;
  font-family: inherit;
  font-size: 2em;
  color: black;
  background: #dfc040;
  cursor: pointer;
  padding: 25px 80px;
  display: inline-block;
  margin: 15px 20px;
  text-transform: uppercase;
  letter-spacing: 1px;
  font-weight: 700;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  width: 500px;
}

.btn:hover {
  color: #0e83cd;
  background: #fff;
}

.daily-trivia-btn {


}

.trivia-game-btn {


}

