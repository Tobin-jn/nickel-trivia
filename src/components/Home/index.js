import React, { Component } from 'react'
import { NavLink } from 'react-router-dom';
import { clearCategory } from '../../actions';
import { connect } from 'react-redux';

export class Home extends Component {

//not sure if this is working, want to clear actegory when we return to the beginning so that we go through the category again
  resetCategory = () => {
    this.props.clearCategory()
  }

  render() {
    return(
      <div>
        <h1 className="title">Nickle Trivia</h1>
        <NavLink className='daily-trivia-btn' to='/categories'>Trivia of the Day</NavLink>
        <NavLink className='trivia-game-btn' to='/categories'>Trivia Game</NavLink>
      </div>
    )
  }
}

export const mapStateToProps = state => ({
  category: state.category
})

export const mapDispatchToProps = dispatch => ({
  clearCategory: () => dispatch(clearCategory())
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Home);