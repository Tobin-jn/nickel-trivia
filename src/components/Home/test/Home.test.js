import React from 'react';
import { shallow } from 'enzyme';
import Home from '../index'

describe('Home', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Home />);
  });

  it("should exist", () => {
    expect(wrapper).toBeDefined();
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});