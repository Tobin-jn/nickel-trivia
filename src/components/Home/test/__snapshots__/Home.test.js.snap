// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="title"
        >
          Welcome to 
          <span
            className="trivia-text"
          >
            Nickel Trivia!
          </span>
        </h1>,
        <img
          alt="Nickel Jar Logo"
          className="nickel-logo"
          src="nickel.png"
        />,
        <div
          className="btn-container"
        >
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/dailyquestion"
          >
            <button
              className="daily-trivia-btn btn"
            >
              Trivia of the Day
            </button>
          </NavLink>
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/triviagame"
          >
            <button
              className="trivia-game-btn btn"
            >
              Trivia Game
            </button>
          </NavLink>
        </div>,
      ],
      "className": "landing-page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Welcome to ",
            <span
              className="trivia-text"
            >
              Nickel Trivia!
            </span>,
          ],
          "className": "title",
        },
        "ref": null,
        "rendered": Array [
          "Welcome to ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Nickel Trivia!",
              "className": "trivia-text",
            },
            "ref": null,
            "rendered": "Nickel Trivia!",
            "type": "span",
          },
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Nickel Jar Logo",
          "className": "nickel-logo",
          "src": "nickel.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/dailyquestion"
            >
              <button
                className="daily-trivia-btn btn"
              >
                Trivia of the Day
              </button>
            </NavLink>,
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/triviagame"
            >
              <button
                className="trivia-game-btn btn"
              >
                Trivia Game
              </button>
            </NavLink>,
          ],
          "className": "btn-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": <button
                className="daily-trivia-btn btn"
              >
                Trivia of the Day
              </button>,
              "to": "/dailyquestion",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Trivia of the Day",
                "className": "daily-trivia-btn btn",
              },
              "ref": null,
              "rendered": "Trivia of the Day",
              "type": "button",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": <button
                className="trivia-game-btn btn"
              >
                Trivia Game
              </button>,
              "to": "/triviagame",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Trivia Game",
                "className": "trivia-game-btn btn",
              },
              "ref": null,
              "rendered": "Trivia Game",
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="title"
          >
            Welcome to 
            <span
              className="trivia-text"
            >
              Nickel Trivia!
            </span>
          </h1>,
          <img
            alt="Nickel Jar Logo"
            className="nickel-logo"
            src="nickel.png"
          />,
          <div
            className="btn-container"
          >
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/dailyquestion"
            >
              <button
                className="daily-trivia-btn btn"
              >
                Trivia of the Day
              </button>
            </NavLink>
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/triviagame"
            >
              <button
                className="trivia-game-btn btn"
              >
                Trivia Game
              </button>
            </NavLink>
          </div>,
        ],
        "className": "landing-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Welcome to ",
              <span
                className="trivia-text"
              >
                Nickel Trivia!
              </span>,
            ],
            "className": "title",
          },
          "ref": null,
          "rendered": Array [
            "Welcome to ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Nickel Trivia!",
                "className": "trivia-text",
              },
              "ref": null,
              "rendered": "Nickel Trivia!",
              "type": "span",
            },
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Nickel Jar Logo",
            "className": "nickel-logo",
            "src": "nickel.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/dailyquestion"
              >
                <button
                  className="daily-trivia-btn btn"
                >
                  Trivia of the Day
                </button>
              </NavLink>,
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/triviagame"
              >
                <button
                  className="trivia-game-btn btn"
                >
                  Trivia Game
                </button>
              </NavLink>,
            ],
            "className": "btn-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": <button
                  className="daily-trivia-btn btn"
                >
                  Trivia of the Day
                </button>,
                "to": "/dailyquestion",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Trivia of the Day",
                  "className": "daily-trivia-btn btn",
                },
                "ref": null,
                "rendered": "Trivia of the Day",
                "type": "button",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": <button
                  className="trivia-game-btn btn"
                >
                  Trivia Game
                </button>,
                "to": "/triviagame",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Trivia Game",
                  "className": "trivia-game-btn btn",
                },
                "ref": null,
                "rendered": "Trivia Game",
                "type": "button",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
